implementations/micrometer-registry-otlp/src/test/java/io/micrometer/registry/otlp/DeltaOtlpMeterRegistryTest.java

```diff
@@ -0,0 +1,347 @@
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.function.Function;
+
+import static io.micrometer.registry.otlp.AggregationTemporality.DELTA;
+import static org.assertj.core.api.Assertions.assertThat;
+
+class DeltaOtlpMeterRegistryTest {
+
+    private static final String METER_NAME = "test.meter";
+
+    private static final String METER_DESCRIPTION = "Sample meter description";
+
+    private static final Tag meterTag = Tag.of("key", "value");
+
+    MockClock clock;
+
+    OtlpConfig otlpConfig = new OtlpConfig() {
+        @Override
+        public AggregationTemporality aggregationTemporality() {
+            return DELTA;
+        }
+
+        @Override
+        public String get(String key) {
+            return null;
+        }
+    };
+
+    OtlpMeterRegistry registry;
+
+    @BeforeEach
+    void init() {
+        clock = new MockClock();
+        registry = new OtlpMeterRegistry(otlpConfig, clock);
+        // Always assume that atleast one step is completed.
+        this.stepOverNStep(1);
+    }
+
+    @Test
+    void counter() {
+        Counter counter = Counter.builder(METER_NAME)
+            .description(METER_DESCRIPTION)
+            .tags(Tags.of(meterTag))
+            .register(registry);
+        counter.increment();
+        counter.increment();
+        assertSum(publishTimeAwareWrite(counter), 0, TimeUnit.MINUTES.toNanos(1), 0);
+        this.stepOverNStep(1);
+        assertSum(publishTimeAwareWrite(counter), TimeUnit.MINUTES.toNanos(1), TimeUnit.MINUTES.toNanos(2), 2);
+
+        this.stepOverNStep(1);
+        counter.increment();
+        assertSum(publishTimeAwareWrite(counter), TimeUnit.MINUTES.toNanos(2), TimeUnit.MINUTES.toNanos(3), 1);
+    }
+
+    @Test
+    void functionCounter() {
+        AtomicLong atomicLong = new AtomicLong(10);
+
+        FunctionCounter counter = FunctionCounter.builder(METER_NAME, atomicLong, AtomicLong::get)
+            .description(METER_DESCRIPTION)
+            .tags(Tags.of(meterTag))
+            .baseUnit("milliseconds")
+            .register(registry);
+
+        assertSum(publishTimeAwareWrite(counter), 0, TimeUnit.MINUTES.toNanos(1), 0);
+        this.stepOverNStep(1);
+        assertSum(publishTimeAwareWrite(counter), TimeUnit.MINUTES.toNanos(1), TimeUnit.MINUTES.toNanos(2), 10);
+        this.stepOverNStep(1);
+        assertSum(publishTimeAwareWrite(counter), TimeUnit.MINUTES.toNanos(2), TimeUnit.MINUTES.toNanos(3), 0);
+    }
+
+    @Test
+    void timer() {
+        Timer timer = Timer.builder(METER_NAME)
+            .description(METER_DESCRIPTION)
+            .tags(Tags.of(meterTag))
+            .register(registry);
+        timer.record(10, TimeUnit.MILLISECONDS);
+        timer.record(77, TimeUnit.MILLISECONDS);
+        timer.record(111, TimeUnit.MILLISECONDS);
+
+        // This is where TimeWindowMax can be painful and make no sense at all. Need to
+        // re-visit this. Probably StepMax is what might fit good for OTLP.
```

Comment:
I think we need a different Max implementation as you say. TimeWindowMax doesn't fit the interval as it should for OTLP.